feat: add resetPassword api

adityasmksaxena opened the pull request:
- add reset password with email token api

adityasmksaxena committed:
refactor: change filenames for standard practices and refactor async tests

adityasmksaxena commented:
![screenshot 2019-03-03 at 10 04 59 pm](https://user-images.githubusercontent.com/5576424/53698288-9b6c5100-3e00-11e9-831a-feae5cf7c0d6.png)

ravikp7 commented:
I have no idea why CI is not running on this pull request. Can someone explain me this?

nimish-gupta commented on server/src/config/index.js:
```diff
@@ -5,10 +5,18 @@ dotenv.config();
   jwt: process.env.JWT_SECRET,
   jwtExp: process.env.JWT_EXPIRY,
 };
+export const serverBaseUrl = `http://localhost:${serverPort}`;
```
Serverbaseurl will be diff  for production servers

adityasmksaxena replied:
It won't be needed, I'm using serverBaseUrl for tests only

nimish-gupta commented on server/src/config/mailer.js:
```diff
@@ -1,6 +1,43 @@
+    <div>Please ignore this message if you didn't signup.</div>
+    </p>`,
+  };
+  sgMail.send(msg);
```
Only send mail when you are in staging or production server, when in local envjust console log, :)

adityasmksaxena replied:
done 👍

nimish-gupta commented on server/src/controllers/auth.controller.js:
```diff
@@ -1,28 +1,17 @@
     const { name, email, password } = req.body;
     const user = await User.createUser({ name, email, password });
     const token = user.generateToken();
     if (!isTest) {
```
Instead of testinghere, set this up in mailer.js, because then you don't have to change your flow in your code to handle trst conditions

adityasmksaxena replied:
sure. thanks :)

nimish-gupta reviewed (changes requested):
Great work:) LGTM after above changes

adityasmksaxena committed:
feat: forgetPassword and resetPassword apis
- POST /auth/forgetPassword
- PUT /api/me/resetPassword
- PUT /api/me/activateUser

fixes #36
